staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/interface.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+// Gauge is the methods of a gauge that are used by instrumented code.
+type Gauge interface {
+	Set(float64)
+	Inc()
+	Dec()
+	Add(float64)
+	SetToCurrentTime()
+}
+
+// RatioedGauge tracks ratios.
+// The numerator is set/changed through the Gauge methods,
+// and the denominator can be updated through the SetDenominator method.
+// A ratio is tracked whenever the numerator or denominator is set/changed.
+type RatioedGauge interface {
+	Gauge
+
+	// SetDenominator sets the denominator to use until it is changed again
+	SetDenominator(float64)
+}
+
+// RatioedGaugeVec creates related observers that are
+// differentiated by a series of label values
+type RatioedGaugeVec interface {
+
```

Comment:
nit: remove this empty line (can be a follow up)

Reply:
done